src/base/result_decoder.re

```diff
@@ -601,13 +638,20 @@ let rec unify_document_schema = (config, document) => {
               Some(fg_selection_set),
             );
 
-          Mod_fragment(
-            fg_name.item,
-            [],
-            error_marker.has_error,
-            fg,
-            structure,
-          );
+          switch (with_decoder) {
+          | Error(err) => Mod_fragment(fg_name.item, [], true, fg, err)
+          | Ok(decoder) =>
+            Mod_fragment(
+              fg_name.item,
+              [],
+              error_marker.has_error,
+              fg,
+              switch (decoder) {
+              | Some(decoder) => decoder(structure)
+              | None => structure
+              },
+            )
+          };
```

Comment:
There are a couple of different ways of doing this, but this seemed most beginner legible to me.